components/Feature/SharePlan/index.js

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react';
+import Table, {
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableBody,
+  TableData
+} from 'components/Table';
+import { Button, Checkbox } from 'components/Form';
+import Heading from 'components/Heading';
+import css from './index.module.scss';
+import ShareStatus from './ShareStatus';
+
+const SharePlan = ({ error, plan, onPlanShared }) => {
+  const [selectedContact, setSelectedContact] = useState({});
+  const [disableShare, setDisableShare] = useState(true);
+
+  const handleSelectEmail = e => {
+    if (!e.target.value) return;
+    const contact = selectedContact;
+    e.target.checked ? (contact.email = e.target.value) : delete contact.email;
```

Comment:
FWIW, if you set `email` to `undefined` when you call `JSON.stringify(contact)` to send it to the API, it will get removed anyway as stringify will ignore any undefined values.

Reply:
Oh that's interesting! It doesn't change the amount of code tho, cause we'd have to set it to undefined so not sure if it's worth changing

Reply:
Yeah don't think we need to change it necessarily, and if the intent is for there to only be either a number _or_ an email then this could even make that clearer.

Or maybe we model it like:
```
setSelectedContact({
  type: "email",
  value: "blah@example.com"
});
```
then we can just replace the whole thing and not worry about needing to remove the old values?